bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: Conditionally set TLS section on ManagedKafka type

## Description

Only set the TLS section for the ManagedKafkaType when external
certificates flag is enabled.

Related to: https://issues.redhat.com/browse/MGDSTRM-3805

## Verification Steps

1. an OCM cluster is needed
1. add it into your local kas fleet manager db
1. edit the secrets/kafka-tls.crt and secrets/kafka-tls.key and put there dummy values
1. start local kas fleet manager DB with the --enable-kafka-external-certificate flag and 
1. create a kafka request assigned it to the ClusterID that has been added to the DB. Then get the ClusterID of the cluster where tha kafka_request has been placed. To get the ID you can take a look at the kafka_request database entry
1. manually call the /api/managed-services-api/v1/agent_clusters/\<id\>/kafkas endpoint. Remember that to call this endpoint the OCM token to provide  has to be the one corresponding to the agent-cluster and not your user's token
1. verify that in the results there is a `.spec.endpoint.tls` section and it contains the `cert` and `key` attributes with the previously set corresponding dummy values

Repeat the previous steps but now when running kas fleet manager DB do NOT set the  --enable-kafka-external-certificate flag.
Then verify that in this case there is NO `.spec.endpoint.tls` at all in the results

## Checklist (Definition of Done)
